src/components/timeline/TrackScreenshots.js

```diff
@@ -58,7 +61,22 @@ class Screenshots extends PureComponent<Props, State> {
     pageX: null,
     containerTop: null,
   };
+  findScreenshotAtMouse(offsetX: number): number | null {
+    const { width, rangeStart, rangeEnd, screenshots } = this.props;
+    const rangeLength = rangeEnd - rangeStart;
+    const mouseTime = (offsetX / width) * rangeLength + rangeStart;
+
+    // Loop backwards to find the latest screenshot that has a time less
+    // than the current time at the mouse position.
+    for (let i = screenshots.length - 1; i >= 0; i--) {
+      const screenshotTime = screenshots[i].start;
+      if (mouseTime >= screenshotTime) {
+        return i;
+      }
+    }
 
+    return null;
+  }
```

Comment:
nit: in general, please keep empty lines between the methods, this makes it easier to read.